staging/src/k8s.io/apimachinery/pkg/conversion/queryparams/convert.go

```diff
@@ -90,7 +90,14 @@ func customMarshalValue(value reflect.Value) (reflect.Value, bool) {
 
 	marshaler, ok := value.Interface().(Marshaler)
 	if !ok {
-		return reflect.Value{}, false
+		if !isPointerKind(value.Kind()) && value.CanAddr() {
```

Comment:
@deads2k [like this](https://play.golang.org/p/kVXnIk2AIg)? If only the pointer type implements `Marshaler`, passing a value type to `customMarshalValue()` causes `CanAddr` to return false.

```
CanAddr reports whether the value's address can be obtained with Addr. Such
values are called addressable. A value is addressable if it is an element of a
slice, an element of an addressable array, a field of an addressable struct, or
the result of dereferencing a pointer. If CanAddr returns false, calling Addr
will panic.
```